NewHaven/src/Resources.cpp

```diff
@@ -210,6 +214,15 @@ Building::~Building()
 	delete buildingNumber;
 }
 
+Building& Building::operator=(const Building& building)
+{
```

Comment:
You should check here to make sure that what you are assigning does not have the same address.

Reply:
Oh true, I didn't put it there because it's not needed, but it would avoid the unnecessary statements to have it there.

Reply:
yeah well in most cases it isnt really needed, but it reduces the level of overhead